pkg/manifests/manifests.go

```diff
@@ -5,239 +5,885 @@ import (
+		},
+		Rules: []rbacv1.PolicyRule{
+			rbacv1.PolicyRule{
+				Verbs: []string{
+					"get",
+					"list",
+					"watch",
+					"create",
+					"update",
+					"patch",
+					"delete",
+				},
+				APIGroups: []string{
+					"cluster.k8s.io",
+				},
+				Resources: []string{
+					"clusters",
+				},
+			},
+			rbacv1.PolicyRule{
+				Verbs: []string{
+					"get",
+					"list",
+					"watch",
+					"create",
+					"update",
+					"patch",
+					"delete",
+				},
+				APIGroups: []string{
+					"cluster.k8s.io",
+				},
+				Resources: []string{
+					"machines",
+				},
+			},
+			rbacv1.PolicyRule{
+				Verbs: []string{
+					"get",
+					"list",
+					"watch",
+					"create",
+					"update",
+					"patch",
+					"delete",
+				},
+				APIGroups: []string{
+					"cluster.k8s.io",
+				},
+				Resources: []string{
+					"machinedeployments",
+				},
+			},
+			rbacv1.PolicyRule{
+				Verbs: []string{
+					"get",
+					"list",
+					"watch",
+					"create",
+					"update",
+					"patch",
+					"delete",
+				},
+				APIGroups: []string{
+					"cluster.k8s.io",
+				},
+				Resources: []string{
+					"machinesets",
+				},
+			},
+			rbacv1.PolicyRule{
+				Verbs: []string{
+					"get",
+					"list",
+					"watch",
+					"create",
+					"update",
+					"patch",
+					"delete",
+				},
+				APIGroups: []string{
+					"cluster.k8s.io",
+				},
+				Resources: []string{
+					"machines",
+				},
+			},
+			rbacv1.PolicyRule{
+				Verbs: []string{
+					"get",
+					"list",
+					"watch",
+					"create",
+					"update",
+					"patch",
+					"delete",
+				},
+				APIGroups: []string{
+					"",
+				},
+				Resources: []string{
+					"nodes",
+				},
+			},
+			rbacv1.PolicyRule{
+				Verbs: []string{
+					"get",
+					"list",
+					"watch",
+					"create",
+					"update",
+					"patch",
+					"delete",
+				},
+				APIGroups: []string{
+					"cluster.k8s.io",
+				},
+				Resources: []string{
+					"machines",
+				},
 			},
 		},
 	}
 }
 
-func ClusterAPIDeployment(clusterAPINamespace string) *appsv1beta2.Deployment {
-	var replicas int32 = 1
-	return &appsv1beta2.Deployment{
+func ClusterRoleBinding(clusterAPINamespace string) *rbacv1.ClusterRoleBinding {
+	return &rbacv1.ClusterRoleBinding{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ClusterRoleBinding",
+			APIVersion: "rbac.authorization.k8s.io/v1",
+		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "clusterapi-apiserver",
+			Name: "manager-rolebinding",
+		},
+		Subjects: []rbacv1.Subject{
+			rbacv1.Subject{
+				Kind:      "ServiceAccount",
+				Name:      "default",
+				Namespace: clusterAPINamespace,
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     "manager-role",
+		},
+	}
+}
+
+func ManagerManifest(clusterAPINamespace, managerImage string) *appsv1.StatefulSet {
+	var terminationGracePeriodSeconds int64 = 10
+	return &appsv1.StatefulSet{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "StatefulSet",
```

Comment:
why does this need to be an statefulset? couldn't it be a deployment

Reply:
Copy of upstream yaml. Not sure what was the reason to change deployment to statefulset.